### ISSUE TITLE
How to upgrade to latest version without waiting for docker.io/mriedmann/humhub

### ISSUE THREAD

**Issue by timaschew** [mriedmann/humhub-docker#230]
I'm using right the latest image `docker.io/mriedmann/humhub:latest` contains HumHub 1.9.1.

But there is already HumHub 1.9.2 (stable) out and there is even a 1.10x as beta.

What do you recommend to do to upgrade to the latest stable (or beta) version of HumHub (not humhub-docker)?

**Comment by ArchBlood** [comment]
I'm actually interested to know if the updater module works with the docker image or not, it would be very helpful in these cases @mriedmann

**Comment by timaschew** [comment]
@ArchBlood no, it doesn't work, see https://github.com/mriedmann/humhub-docker/issues/227
This happens even when trying to upgrade from 1.9.1 to 1.9.2 with the updater module.

**Comment by ArchBlood** [comment]
> @ArchBlood no, it doesn't work, see #227 This happens even when trying to upgrade from 1.9.1 to 1.9.2 with the updater module.

Could you provide what version of PHP is used? As of now from 1.9.x+, the PHP requirements have been increased so this could be a problem 🤔

Overall it seems related to this, that in of itself could explain why `rename()` isn't working
https://stackoverflow.com/questions/54834167/docker-compose-mounted-directory-does-not-rename-folders

**Comment by timaschew** [comment]
I'm using the **AllInOne* variant docker image. The dockerfile shows PHP 7: https://github.com/mriedmann/humhub-docker/blob/master/Dockerfile

**Comment by ArchBlood** [comment]
> I'm using the *_AllInOne_ variant docker image. The dockerfile shows PHP 7: https://github.com/mriedmann/humhub-docker/blob/master/Dockerfile

That could be the issue see the current version of PHP HumHub runs on is PHP 7.3+, if your dockerfile is running PHP 7 then it needs updated, or at least the package needs updated 🤔
https://github.com/humhub/humhub/blob/master/composer.json#L16

**Comment by ArchBlood** [comment]
I'd suggest using `php74-cli` or just `php74`

**Comment by timaschew** [comment]
Would be nice if @mriedmann could update this in the Dockerfile.

**Comment by timaschew** [comment]
@ArchBlood I don't thinks that's the problem.

If I run this command in the docker container it shows me this:

```
# php --version
PHP 7.4.23 (cli) (built: Aug 26 2021 23:05:01) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
```

**Comment by timaschew** [comment]
I've published a docker image for HumHub 1.9.2 via [my own repository](https://gitlab.com/timaschew/humhub-docker/-/blob/master/.gitlab-ci.yml) now.

`registry.gitlab.com/timaschew/humhub-docker:1.9.2`

**Comment by timaschew** [comment]
I've published a docker image for HumHub 1.9.2 via [my own repository](https://gitlab.com/timaschew/humhub-docker/-/blob/master/.gitlab-ci.yml) now.

`registry.gitlab.com/timaschew/humhub-docker:1.9.2`

**Comment by timaschew** [comment]
from https://github.com/mriedmann/humhub-docker/issues/227#issuecomment-946176589

> Hey!
> 
> The normal upgrade-path would be, restarting a container with an upgradable version (e.g. 1.9.x) using the same directories. In theory the entrypoint-script should run the proper upgrade tasks (see https://github.com/mriedmann/humhub-docker/blob/master/docker-entrypoint.sh#L100).

Indeed, that worked.
First I run `docker network ls` to find the network id of `humhub-test_default` because otherwise it's not possible to establish a connection to the database. 
Then I've added the volumes (not sure if all of them are required), which are:

- /var/www/localhost/htdocs/protected/config
- /var/www/localhost/htdocs/protected/modules
- /var/www/localhost/htdocs/uploads
- var/www/localhost/htdocs/assets 

```
docker run -it  -v /host:/container --network 33c2decda3e8 registry.gitlab.com/timaschew/humhub-docker:1.9.2
```

This was the output:

```
/docker-entrypoint.sh: Starting pre-launch ...
/docker-entrypoint.sh: Existing installation found!
db (172.20.0.2:3306) open
/docker-entrypoint.sh: Updating from version v1.9.1 to v1.9.2
Yii Migration Tool (based on Yii v2.0.40)
No new migrations found. Your system is up-to-date.
Rebuild search index: .OK!
'/usr/src/humhub/.version' -> '/var/www/localhost/htdocs/protected/config/.version'
/docker-entrypoint.sh: Config preprocessing ...
/docker-entrypoint.sh: installation active
/docker-entrypoint.sh: debug disabled
/docker-entrypoint.d/50-humhub-validate.sh
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/50-humhub-validate.sh
validating ...

*** Performing database integrity checks

Validating: File Module (0 entries)
Validating: Activity Module (0 entries)
Validating: Space Module - Spaces (0 entries)
Validating: Space Module - Memberships (0 entries)
Validating: Like (0 entries)
Validating: User Module - ContentContainer (1 entries)
Validating: User Module - Users (1 entries)
Validating: User Module - Password (1 entries)
Validating: User Module - Profile (1 entries)
Validating: User Module - Mentioning (0 entries)
Validating: User Module - Follow (0 entries)
Validating: User Module - Content container (1 entries)
Validating: Notification Module (0 entries)
Validating: Content Objects (0 entries)
Validating: Comment Module (0 entries)
Validating: Post  Module - Posts (0 entries)

*** All integrity checks done

/docker-entrypoint.sh: Launching /docker-entrypoint.d/60-nginx-config.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/70-nginx-allowedips.sh
Setting HUMHUB_REVERSEPROXY_WHITELIST
Added 127.0.0.1 to Reverseproxy Whitelist
/docker-entrypoint.sh: Configuration complete; ready for start up
/docker-entrypoint.sh: Entrypoint finished! Launching ...
tail: can't open '/var/www/localhost/htdocs/protected/runtime/logs/app.log': No such file or directory
tail: no files
2021/11/02 15:48:31 [notice] 39#39: using the "epoll" event method
2021/11/02 15:48:31 [notice] 39#39: nginx/1.20.1
2021/11/02 15:48:31 [notice] 39#39: OS: Linux 5.10.47-linuxkit
2021/11/02 15:48:31 [notice] 39#39: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2021/11/02 15:48:31 [notice] 39#39: start worker processes
2021/11/02 15:48:31 [notice] 39#39: start worker process 41
[02-Nov-2021 15:48:31] NOTICE: fpm is running, pid 40
[02-Nov-2021 15:48:31] NOTICE: ready to handle connections
```
